### PR TITLE
fraggenescan 1.31 (new formula)

### PR DESCRIPTION
**Pull request by suujia**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by suujia** [comment]
Readme: 

```
1. Untar the downloaded file "FragGeneScan.tar.gz". This will automatically generate the directory "FragGeneScan".

2. Make sure that you also have a C compiler such as "gcc" and perl interpreter.

3. Run "makefile" to compile and build excutable "FragGeneScan"
	make clean
	make fgs
```

**Comment by sjackman** [comment]
> Empty installation

```ruby
bin.install "fgs"
```

**Comment by tseemann** [comment]
```
==> Summary
/home/linuxbrew/.linuxbrew/Cellar/fraggenescan/1.31: 6 files, 82.5KB
==> brew install --only-dependencies --include-test brewsci/bio/fraggenescan
==> Installing fraggenescan from brewsci/bio
All dependencies for brewsci/bio/fraggenescan are satisfied.
==> brew test brewsci/bio/fraggenescan --verbose
Testing brewsci/bio/fraggenescan
==> /home/linuxbrew/.linuxbrew/Cellar/fraggenescan/1.31/bin/run-help
Error: brewsci/bio/fraggenescan: failed
No such file or directory - /home/linuxbrew/.linuxbrew/Cellar/fraggenescan/1.31/bin/run-help
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula_assertions.rb:9:in ``'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula_assertions.rb:9:in `shell_output'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/brewsci/homebrew-bio/Formula/fraggenescan.rb:23:in `block in <class:Fraggenescan>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1693:in `block (3 levels) in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:559:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1692:in `block (2 levels) in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:861:in `with_logging'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1691:in `block in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1928:in `block in mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `block in run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `chdir'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1927:in `mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1685:in `run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb:28:in `block in <main>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
```

**Comment by suujia** [comment]
Thanks! I got it working with 
```
prefix.install Dir["*"]
bin.install_symlink Dir["#{prefix}/run_FragGeneScan.pl"]
```

File: Formula/fraggenescan.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "perl"
+
+  def install
+    system "make", "clean"
+    prefix.install Dir["*"]
+    bin.install_symlink Dir["#{prefix}/run_FragGeneScan.pl"]
+  end
+
+  test do
+    assert_match "ERROR", shell_output("#{bin}/run_FragGeneScan.pl 2>&1")
```
Comment (sjackman) [inline Formula/fraggenescan.rb]: Please change `ERROR` to `USAGE`.

File: Formula/fraggenescan.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Fraggenescan < Formula
+  # cite Rho_2010: "https://doi.org/10.1093/nar/gkq747"
+  desc "Predicting genes in short and error-prone reads"
+  homepage "https://github.com/COL-IU/FragGeneScan"
+  url "https://downloads.sourceforge.net/project/fraggenescan/FragGeneScan1.31.tar.gz"
+  sha256 "cd3212d0f148218eb3b17d24fcd1fc897fb9fee9b2c902682edde29f895f426c"
+
+  depends_on "gcc"
```
Comment (sjackman) [inline Formula/fraggenescan.rb]: This dependency on `gcc` shouldn't be necessary. is it addressing a particular error that you're seeing?

File: Formula/fraggenescan.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "perl"
```
Comment (sjackman) [inline Formula/fraggenescan.rb]: On macOS we use the host `perl`.
```ruby
depends_on "perl" unless OS.mac?
```

**Comment by suujia** [comment]
@sjackman  Thanks! Yes, still green after those changes 👍

**Comment by sjackman** [comment]
Merged. Thanks, Susie! 💖

**Pull request closed without merging**